Added `engine.js` documentation

gilkeidar opened the pull request:
Added JSDocs documentation to `engine.js` and removed commented out code.

Closes #35

gilkeidar committed:
Added `engine.js` documentation

Added JSDocs documentation to `engine.js` and removed commented out code.

gilkeidar committed:
Merge branch 'main' into engine-documentation

minhnhat1901 reviewed (commented):
After checking the code carefully, I think everything looks good and we can merge to the main branch and work on the next sprint

gilkeidar committed:
Slight Changes to FortuneEngine Code

1. db_reader now fetches from the given path string instead of an assumed path
2. Added an export statement so that the FortuneEngine class can be used in separate files.

gilkeidar commented:
Slight changes to the `FortuneEngine` class:

1. `db_reader` now reads the JSON file from the given path instead of constructing an expected path. This expected path did not work when trying to work from the `mini-apps` directory.
2. Added an `export` statement at the end of the file to export the `FortuneEngine` class so that it can be used in other files.
